Userland/Applications/HexEditor/HexEditor.cpp

```diff
@@ -623,3 +623,37 @@ Vector<Match> HexEditor::find_all(ByteBuffer& needle, int start)
 
     return matches;
 }
+
+Vector<Match> HexEditor::find_all_strings(size_t min_length)
+{
+    if (m_buffer.is_empty())
+        return {};
+
+    Vector<Match> matches;
+
+    int offset = -1;
+    StringBuilder builder;
+    for (size_t i = 0; i < m_buffer.size(); i++) {
+        char c = m_buffer.bytes().at(i);
+        if (isprint(c)) {
+            if (offset == -1)
+                offset = i;
+            builder.append(c);
+        } else {
+            if (builder.length() >= min_length) {
+                dbgln("find_all_strings: relative_offset={} string={}", offset, builder.to_string());
+                matches.append({ offset, builder.to_string().characters() });
```

Comment:
Sounds to me like this is leaking pointers to invalid data?

Reply:
or does `Match` contain a `String` and not a `StringView`? in which case, this is a copy _and_ a strlen, let's at least avoid that.

Reply:
> 
> 
> or does `Match` contain a `String` and not a `StringView`? in which case, this is a copy _and_ a strlen, let's at least avoid that.

`Match` contain a `String` not a `StringView` although in retrospect it should have been a `ByteBuffer`, as the string conversion can be performed by the model.

I presume you're suggesting to drop `.characters()`.

Reply:
Yeah, `.characters()` is not needed, let's drop that.